src/main.rs

```diff
@@ -11,7 +11,11 @@ use structopt::StructOpt;
 fn main() -> std::io::Result<()> {
     env_logger::init();
 
-    let matches = cli::ParsecToolApp::from_args();
+    let mut matches = cli::ParsecToolApp::from_args();
+    matches.init_client().map_err(|e| {
+        err!("{:?}", e);
+        std::io::Error::new(std::io::ErrorKind::Other, "Spinning up BasicClient failed.")
+    })?;
     matches.subcommand.run(&matches).map_err(|e| {
```

Comment:
I think it's very good to spin up the BasicClient before but couldn't it be cleaner if we do it directly in the `main` function and pass the reference to the `run` method? It seems cleaner than having it in the structopt structure, even though we can skip it. That way we keep both things separate (CLI and client)

Reply:
Sure, that works for me - we can pass the whole client like that, not just a reference, since we only run one subcommand anyway. I'll update